xcode/Subconscious/Shared/Components/AddressBook/AddFriendViaQRCodeView.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  AddFriendViaQRCodeView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 6/3/2023.
+//
+
+import Foundation
+import SwiftUI
+import CodeScanner
+import os
+
+struct AddFriendViaQRCodeView: View {
+    @Environment(\.dismiss) var dismiss
+    
+    var onScannedDid: (String) -> Void
+    var onCancel: () -> Void
+    
+    static let logger = Logger(
+        subsystem: Config.default.rdns,
+        category: "AddFriendViaQRCodeView"
+    )
+    
+    var body: some View {
+        VStack {
+            // Toolbar is unavailable here, create a minimal one that blends with notch/bezel.
+            HStack {
+                Button("Cancel", role: .cancel) {
+                    dismiss()
+                }
+                .buttonStyle(.borderless)
+
+                Spacer()
+            }
+            .padding(AppTheme.unit2)
+            .background(Color.black)
+            
+            CodeScannerView(
+                codeTypes: [.qr],
+                showViewfinder: true,
+                simulatedData: "did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7",
+                shouldVibrateOnSuccess: true
```

Comment:
Is vibrate on success a typical pattern for this kind of interaction?

Reply:
I can't find any mention in the Apple HIG about this, I was going off of my own personal preference while using it. The view dismisses itself _as soon as_ a code is enters the view which can be a little shocking for the user.

Other option is a little toast notification of some kind saying the scan succeeded?

Reply:
SGTM. Let's move forward with what you have here. We can always refine later.